src/hooks/useHasAppUpdated/useHasAppUpdated.ts

```diff
@@ -1,42 +1,54 @@
 import { useInterval } from '@chakra-ui/hooks'
 import axios from 'axios'
 import isEqual from 'lodash/isEqual'
-import { useCallback, useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 
-const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
+export const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
 
 export const useHasAppUpdated = () => {
   const [hasUpdated, setHasUpdated] = useState(false)
-  const [initialManifestMainJs, setInitialManifestMainJs] = useState(null)
-  const [initialEnvMainJs, setInitialEnvMainJs] = useState(null)
-
-  const fetchData = async (url: string) => {
-    try {
-      const { data } = await axios.get(url)
-      return data
-    } catch (e) {
-      return console.error(`useHasAppUpdated: error fetching data from URL: ${url}`, e)
-    }
-  }
-
-  // 'asset-manifest.json' keeps track of the latest minified built files.
-  // interpolated with a dummy query param to bypass the browser cache.
-  const assetManifestUrl = `/asset-manifest.json?${new Date().valueOf()}`
-  const storeMainManifestJs = async () => {
+  const [initialManifestMainJs, setInitialManifestMainJs] = useState<unknown>()
+  const [initialEnvMainJs, setInitialEnvMainJs] = useState<unknown>()
+
+  const isLocalhost = window.location.hostname === 'localhost'
+
+  const fetchData = useCallback(
+    async (url: string): Promise<unknown> => {
+      // don't ever try to fetch on localhost - we don't care
+      if (isLocalhost) return {} // need to return dummy value
+      try {
+        // dummy query param to bypass the browser cache.
+        const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
+        return data
+      } catch (e) {
+        console.error(`useHasAppUpdated: error fetching data from URL: ${url}`, e)
+      }
+    },
+    [isLocalhost],
+  )
+
+  // 'asset-manifest.json' keeps track of the latest minified built files
+  const assetManifestUrl = `/asset-manifest.json`
+  const storeMainManifestJs = useCallback(async () => {
     const manifestMainJs = await fetchData(assetManifestUrl)
-    setInitialManifestMainJs(manifestMainJs)
-  }
-  useCallback(storeMainManifestJs, [assetManifestUrl])
+    manifestMainJs && setInitialManifestMainJs(manifestMainJs)
+  }, [assetManifestUrl, fetchData])
 
-  // interpolated with a dummy query param to bypass the browser cache.
-  const envUrl = `/env.json?${new Date().valueOf()}`
-  const storeMainEnvJs = async () => {
+  // 'asset-manifest.json' keeps track of the current environment variables
+  const envUrl = `/env.json`
```

Comment:
what's the justification for removing the cache-busting query parameter (or the other way around, why was the query parameter added in the first place)?

Reply:
it just got moved to a common location above as the code was duplicated https://github.com/shapeshift/web/pull/1519/files#diff-cdfbf840a246002bef3e005f6913c541ff0ac8bdb89f90b597be89220469ced1R21